### ISSUE TITLE
Feature request: 64-bit integer type on 32-bit systems

### ISSUE THREAD

**Issue by data-man** [albertodemichelis/squirrel#64]
It is really very necessary.

**Comment by zeromus** [comment]
This could mean one of two things. 
1. Add a completely new longer-integer type (not gonna happen, I feel I can safely say)
2. Make SQInteger 64bits.  Actually, I have recently done the opposite--using 32bit integers on a 64bit system. This necessitated some cleanup to the typedef system. I'll investigate whether I can also incorporate an option to increase integers to 64bit.

**Comment by mfxmfx** [comment]
I agree that this would be **very** useful, but it seems some work and will probably break lots of
3rd party modules, so I guess this will not happen before the next major version.

The first step would be using ```size_t```, ```ssize_t``` and ``ptrdiff_t`` instead of SQInteger in a lot of places so that the host (size_t) and the target  (SQInteger) integer types are correctly decoupled.

**Comment by zeromus** [comment]
https://github.com/zeromus/squirrel/commits/32-on-64
You can see my work here for running a 32bit SQInteger on a 64bit host. Maybe I'm only lucky that this seems to work for me, but I didnt find any places that needed that decoupling you mentioned. It seems the author generally uses SQInteger in place of 'int' which is annoying, but rather benign. Could you give an example of a place you think there's trouble?

**Comment by bluebear94** [comment]
I'd like this to happen as well. Do we have any update?

**Comment by EternityForest** [comment]
SQInteger is used in a million and one places. Now I'm all for making the Squirrel language integer type 64 bits, but SQInteger is used for things like measuring stack entries in a lot of code I've seen, and really we don't need to be wasting any more RAM than needed.

But 64 bit ints in the language seem like a great idea. It would allow microseconds-since-epoch timestamps, and in an intepreted language the extra 4 bytes per int object don't seem all that relevant compared to all the object management overhead.

What about having a separate SQLangInt type that determines the width of the integers created from within squirrel?
